### PR TITLE
JS Components: Convert AutomatticBylineLogo component to TypeScript

### PR DESCRIPTION
**Pull request by manzoorwanijk**
#### Changes proposed in this Pull Request:
* This PR converts `AutomatticBylineLogo` component to TypeScript and cleans up the propTypes.
* It also converts the `AutomatticBylineLogo` stories and unit tests to TS
* Adds more test cases for `AutomatticBylineLogo` unit tests

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `AutomatticBylineLogo` component work as expected
* Check the consumers of `AutomatticBylineLogo` and related components and confirm that they work as expected
* Check for any regression

Screenshot 2022-07-04 at 2 05 56 PM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202276314263996

### CODE REVIEW

File: projects/js-packages/components/components/automattic-byline-logo/types.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type React from 'react';
+
+export interface AutomatticBylineLogoProps extends React.SVGProps< SVGSVGElement > {
+	/**
+	 * Title for SVG.
+	 */
+	title?: string;
+
+	/*
+     Height for SVG. */
```
Comment (renatoagds) [inline projects/js-packages/components/components/automattic-byline-logo/types.ts]: Small adjustment here

Comment (manzoorwanijk) [reply projects/js-packages/components/components/automattic-byline-logo/types.ts]: Fixed. Thanks

File: projects/js-packages/components/components/automattic-byline-logo/test/component.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+			render( <AutomatticBylineLogo /> );
+
+			expect( screen.getByText( 'An Automattic Airline' ) ).toBeInTheDocument();
+		} );
+
+		it( 'renders the given title', () => {
+			render( <AutomatticBylineLogo title="An A8C Airline" /> );
+
+			expect( screen.getByText( 'An A8C Airline' ) ).toBeInTheDocument();
+		} );
```
Comment (dhasilva) [inline projects/js-packages/components/components/automattic-byline-logo/test/component.tsx:26]: :clap:

**Review by dhasilva — DISMISSED** [review]
:rocket: 
Screenshot_2022-07-26_15-37-31

**Review by retrofox — APPROVED** [review]
LGTM

**Comment by manzoorwanijk** [comment]
r249673-wpcom

**Pull request merged**